pandas/_libs/src/inference.pyx

```diff
@@ -218,9 +218,90 @@ cdef _try_infer_map(v):
     return None
 
 
-def infer_dtype(object _values):
+def infer_dtype(object value):
     """
-    we are coercing to an ndarray here
+    Effeciently infer the type of a passed val, or list-like
+    array of values. Return a string describing the type.
+
+    Parameters
+    ----------
+    value : scalar, list, ndarray, or pandas type
+
+    Returns
+    -------
+    string describing the common type of the input data.
+    Results can include:
+       - floating
```

Comment:
done, though I don't think we actually have a doc-generation section for ``pandas.api`` ATM. (i'll make an issue).